src/OpenTelemetry/Trace/Configuration/OpenTelemetrySdk.cs

```diff
@@ -106,5 +100,35 @@ public static void EnableOpenTelemetry(Action<OpenTelemetryBuilder> configureOpe
 
             ActivitySource.AddActivityListener(listener);
         }
+
+        internal static void BuildSamplingParameters(
+            in ActivityCreationOptions<ActivityContext> options, out ActivitySamplingParameters samplingParameters)
+        {
+            ActivityContext parentContext = options.Parent;
+            if (options.Parent == default)
+            {
+                // Check if there is already a parent for the current span.
+                var parentSpan = Activity.Current;
+                if (parentSpan != null)
+                {
+                    parentContext = parentSpan.Context;
+                }
+            }
+
+            // This is not going to be the final traceId of the Activity (if one is created), however, it is
+            // needed in order for the sampling to work. This differs from other OTel SDKs in which it is
+            // the Sampler always receives the actual traceId of a root span/activity.
+            ActivityTraceId traceId = parentContext.TraceId != default
+                ? parentContext.TraceId
+                : ActivityTraceId.CreateRandom();
```

Comment:
If this Activity is sampled in based on traceId1 (generated randomly), but the actual Activity.TraceId is traceid2, then all the child activities will be having traceid2 and they will be sampled based on traceid2 - this means they could be not sampled in, when the parent was sampled in. Do you see this as an issue? Or this is *not* an issue, as even probability sampler looks at parent sampling flags first to make sampling decision?

Reply:
Let's have this conversation at https://github.com/open-telemetry/opentelemetry-dotnet/pull/702#discussion_r431951369